core/src/main/java/com/radixdlt/genesis/olympia/converter/resource/OlympiaNonXrdConverter.java

```diff
@@ -0,0 +1,309 @@
+    return tuple(resourcesChunks, resourceBalancesChunks);
+  }
+
+  private static BigInteger[] olympiaTotalSupplies(
+      List<OlympiaStateIR.Resource> resources,
+      List<OlympiaStateIR.AccountBalance> olympiaNonXrdBalances) {
+    // Just a note: resourceTotalSuppliesOnOlympia includes a spot
+    // for XRD, but it's always null (because olympiaNonXrdBalances doesn't contain
+    // and balances where resourceIndex == xrdIndex).
+    // When we later use those values, we never use XRD's index.
+    final BigInteger[] resourceTotalSuppliesOnOlympia = new BigInteger[resources.size()];
+    for (var balance : olympiaNonXrdBalances) {
+      final var resIdx = balance.resourceIndex();
+      if (resourceTotalSuppliesOnOlympia[resIdx] == null) {
+        resourceTotalSuppliesOnOlympia[resIdx] = balance.amount();
+      } else {
+        resourceTotalSuppliesOnOlympia[resIdx] =
+            resourceTotalSuppliesOnOlympia[resIdx].add(balance.amount());
+      }
+    }
+    return resourceTotalSuppliesOnOlympia;
+  }
+
+  private static Decimal scaleResourceAmount(
+      BigInteger originalAmount,
+      BigInteger resourceTotalSupplyOnOlympia,
+      BigInteger resourceMaxSupplyOnBabylon) {
+    if (resourceTotalSupplyOnOlympia.compareTo(resourceMaxSupplyOnBabylon) <= 0) {
+      // No need to scale, guaranteed to fit
+      return Decimal.unsafeFromRawBigIntRepr(originalAmount);
+    } else {
+      // Scale it down, using integer div rounding
+      final var scaledBigInt =
+          resourceMaxSupplyOnBabylon.multiply(originalAmount).divide(resourceTotalSupplyOnOlympia);
+      return Decimal.unsafeFromRawBigIntRepr(scaledBigInt);
+    }
+  }
+
+  private static ImmutableList<GenesisDataChunk.Resources> prepareResourcesChunks(
+      OlympiaToBabylonConverterConfig config,
+      List<OlympiaStateIR.Account> accounts,
+      List<OlympiaStateIR.Resource> resources,
+      int olympiaXrdResourceIndex,
+      Decimal[] resourceTotalSuppliesOnBabylon) {
+    return createChunks(
+        resources,
+        config.maxResourcesPerChunk(),
+        (idx, olympiaResource) -> {
+          if (idx == olympiaXrdResourceIndex) {
+            // skip XRD
+            return Optional.empty();
+          }
+          final var initialSupply =
+              resourceTotalSuppliesOnBabylon[idx] == null
+                  ? Decimal.ZERO
+                  : resourceTotalSuppliesOnBabylon[idx];
+          return Optional.of(convertResource(accounts, initialSupply, olympiaResource));
+        },
+        GenesisDataChunk.Resources::new);
+  }
+
+  private static GenesisResource convertResource(
+      List<OlympiaStateIR.Account> accounts,
+      Decimal initialSupply,
+      OlympiaStateIR.Resource resource) {
+    final var metadata =
+        ImmutableList.of(
+            tuple("symbol", resource.symbol()),
+            tuple("name", resource.name()),
+            tuple("description", resource.description()),
+            tuple("url", resource.url()),
```

Comment:
This should be `info_url`.

Actually, more generally, both `info_url` and `icon_url` should be of metadata type URL, according to the metadata standard - so we will need to make the metadata map typed.